### PR TITLE
fix: support set custom nonce to zero

### LINKED ISSUE #10703: [Bug]: Custom nonce doesn't allow for 0 as input
### Describe the bug

Report came from CS

> when editing the custom nonce on mobile, it doesn't allow you to go to 0. The lowest it will go is 1.
> User has a stuck transaction on nonce 0, and is unable to replace it.

> Pasting in 0 doesn't work either I guess. It just auto defaults to the next usable nonce. It works initially, but when you actually submit it, it will be submitted as the next available nonce.

### Expected behavior

_No response_

### Screenshots/Recordings

_No response_

### Steps to reproduce

1. Start a send transaction
2. Try to edit the nonce to 0

### Error messages or log output

_No response_

### Detection stage

In production (default)

### Version

7.28

### Build type

None

### Device

iphone

### Operating system

iOS

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**

This PR adds support for setting the custom nonce to zero and addressing an issue reported by Customer Support.

**Context**
Reported by: CS
Problem: Users are unable to set the custom nonce to 0 on mobile devices. The lowest value currently allowed is 1, which prevents users from replacing transactions stuck at nonce 0.
Behavior: When attempting to set the nonce to 0, either through manual input or by pasting, the input initially appears to accept the value, but on submission, the nonce is automatically adjusted to the next available value, bypassing 0 entirely.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/10703

## **Manual testing steps**

1. Start a send transaction
2. Try to edit the nonce to 0

### **After**

[Screencast from 21-08-2024 14:47:22.webm](https://github.com/user-attachments/assets/0daa47f0-bdb7-4260-b352-bc4e2e6f6fdc)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/confirmations/SendFlow/components/CustomNonceModal/index.js [hunk 1]
```diff
@@ -122,9 +122,11 @@ const CustomModalNonce = ({ proposedNonce, nonceValue, close, save }) => {
   const styles = createStyles(colors);
 
   const incrementDecrementNonce = (decrement) => {
-    let newValue = nonce;
-    newValue = decrement ? --newValue : ++newValue;
-    onChangeText(newValue > 1 ? newValue : 1);
+    const currentNonce = Number(nonce);
+    const updatedValue = decrement ? currentNonce - 1 : currentNonce + 1;
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/SendFlow/components/CustomNonceModal/index.js:126]: Minor, but would calling `isDecrement` make it more explicit that it's a boolean?

File: app/components/Views/confirmations/SendFlow/components/CustomNonceModal/index.test.tsx [hunk 2]
```diff
@@ -2,25 +2,44 @@ import React from 'react';
 import { shallow } from 'enzyme';
 import CustomNonceModal from '.';
 
+const proposedNonce = 26;
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/SendFlow/components/CustomNonceModal/index.test.tsx:5]: Minor, but do we try do uppercase and underscores for all constants?

**Review by matthewwalsh0 — DISMISSED** [review]

File: app/components/Views/confirmations/SendFlow/components/CustomNonceModal/index.js (hunk 1 above)
Comment (vinistevam) [reply app/components/Views/confirmations/SendFlow/components/CustomNonceModal/index.js]: Done [here](https://github.com/MetaMask/metamask-mobile/pull/10729/commits/9f4c891e7319702f3d60b8625800da77b517f66b).

**Comment by sleepytanya** [comment]
Bitrise build https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/b68ef483-3892-4ee7-8a90-b9ef344c863e?tab=artifacts

Nonce can be successfully set to zero:

iOS


https://github.com/user-attachments/assets/6863c88f-1c72-4aa2-9f41-063f0ecc0844

Android


https://github.com/user-attachments/assets/22034d00-099c-414f-a957-fc1094ff07af

File: app/components/Views/confirmations/SendFlow/components/CustomNonceModal/index.js [hunk 3]
```diff
@@ -121,10 +121,12 @@ const CustomModalNonce = ({ proposedNonce, nonceValue, close, save }) => {
   const { colors, themeAppearance } = useTheme();
   const styles = createStyles(colors);
 
-  const incrementDecrementNonce = (decrement) => {
-    let newValue = nonce;
-    newValue = decrement ? --newValue : ++newValue;
-    onChangeText(newValue > 1 ? newValue : 1);
+  const incrementDecrementNonce = (isDecrement) => {
```
Comment (digiwand) [inline app/components/Views/confirmations/SendFlow/components/CustomNonceModal/index.js:124]: what if we updated `incrementDecrementNonce` → `incrementOrDecrementNonce` to be more explicit?

**Review by digiwand — APPROVED** [review]
Great updates with the fix! 

I left 1 non-blocking suggestion to update a function name that was named before this PR

**Review by jpuri — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

File: app/components/Views/confirmations/SendFlow/components/CustomNonceModal/index.js (hunk 3 above)
Comment (vinistevam) [reply app/components/Views/confirmations/SendFlow/components/CustomNonceModal/index.js:124]: That is a great suggestion. Can I do it separately? This PR has been open for a while.

**Pull request merged**